test/environment/create-connector-spec.js

```diff
@@ -0,0 +1,21 @@
+import { createBlueConnector } from '@locker/near-membrane-base';
+
+/*
+    These tests are exercising the BUILT near-membrane-base,
+    ie. package/near-membrane-base/dist/index.js
+*/
+describe('createBlueConnector', () => {
+    it('creates a different connector for the same window object with a different location', (done) => {
+        expect.assertions(1);
+        const iframe = document.createElement('iframe');
+        document.body.appendChild(iframe);
+        const connector1 = createBlueConnector(iframe.contentWindow);
+        iframe.src = 'about:blank';
+        setTimeout(() => {
```

Comment:
@rwaldron how do I test without setTimeout?

Reply:
Fun!

```
    it('creates a different connector for the same window object with a different location', async () => {
        expect.assertions(1);
        const iframe = document.createElement('iframe');
        document.body.appendChild(iframe);
        const connector1 = createBlueConnector(iframe.contentWindow);
        iframe.src = 'about:blank';

        let resolver;
        const promise = new Promise(() => {
          resolver = resolve;
        });
        iframe.onload = () => {
          const connector2 = createBlueConnector(iframe.contentWindow);
          expect(connector1).not.toBe(connector2);
          iframe.remove();
          resolver();
        };
        await promise;
    });

```

This is close to the "deferred pattern", but instead of letting some consumer have our promise, we're resolving it ourselves.